### ISSUE TITLE
data points are corrupted by adjacent Inf points and become NaN

### ISSUE THREAD

**Issue by belm0** [VictoriaMetrics/VictoriaMetrics#805]
**Describe the bug**
Import some data points (integer and inf).
Read back the data points:  some of the integers become `NaN`.

**To Reproduce**
```
$ curl -d 'foo 123' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -d 'foo inf' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -d 'foo -inf' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -d 'foo 456' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,456],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073]}
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,NaN],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073]}
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
$ curl -d 'foo 789' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,NaN,789],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073,1601613299571]}
```
and more trouble...
```
$ curl -d 'foo 1001' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,NaN,789,1001],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073,1601613299571,1601613815109]}
$ curl -d 'foo inf' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,NaN,789,1001,+Inf,-Inf],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073,1601613299571,1601613815109,1601613827435,1601613845394]}
$ curl -d 'foo 1002' -X POST 'http://localhost:8428/api/v1/import/prometheus'
$ curl -G 'http://localhost:8428/api/v1/export' -d 'match[]=foo{}'
{"metric":{"__name__":"foo"},"values":[NaN,+Inf,-Inf,NaN,NaN,NaN,+Inf,-Inf,1002],"timestamps":[1601613197848,1601613207829,1601613214585,1601613221073,1601613299571,1601613815109,1601613827435,1601613845394,1601613860657]}
```
**Version**
```
victoria-metrics-20200929-211927-tags-v1.42.0-0-ge59de9838
```

**Used command-line flags**
```
$ docker run -it --rm -v ~/tmp/vmetrics_data:/victoria-metrics-data -p 8428:8428 victoriametrics/victoria-metrics
```

**Comment by belm0** [comment]
it seems directly related to the `inf` data points

**Comment by valyala** [comment]
@belm0 , thanks for catching the bug! It should be fixed in the commit bc7d67cee22d54ef389b258a21e0f0977b987afa

**Comment by valyala** [comment]
The bug should be fixed starting from [release v1.43.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
